cmd/main.go

```diff
@@ -434,7 +436,35 @@ func autoMigrationWithLeader(migrationLeader leader.ElectorInterface, db *gorm.D
 }
 
 func uploadBaseISOWithLeader(uploadLeader leader.ElectorInterface, objectHandler s3wrapper.API, generator generator.ISOInstallConfigGenerator, log logrus.FieldLogger) error {
+
 	ctx := context.Background()
+
+	// First, we must determine our base object
+	if Options.RHCOSBaseImage == "" {
+		s3wrapper.BaseObjectName = s3wrapper.RHCOSBaseObjectName
+	} else {
+		// Need to get the volume id from the ISO provided
+		iso, err := os.Open(Options.RHCOSBaseImage)
+		if err != nil {
+			log.WithError(err).Fatal("Failed to open provided base image for inspection")
+			return err
+		}
+		defer iso.Close()
+
+		// Need a method to identify the ISO provided
+		// Based on the ISO 9660 standard we should consistently be able to
+		// grab the Volume ID here.
+		volumeId := make([]byte, 32)
+		_, err = iso.ReadAt(volumeId, 32808)
+		if err != nil {
+			log.WithError(err).Fatal("Failed to read volume id from provided base image")
+			return err
+		}
+		// TODO: Should we verify that the volume id is of the form `rhcos-<version>`?
+		s3wrapper.BaseObjectName = strings.TrimSpace(string(volumeId)) + ".iso"
```

Comment:
Is there a more clear way we could do this?
Maybe by using [this iso package](https://github.com/kdomanski/iso9660/blob/880de9cb649c5731170a2f808b11ef1392954864/image_reader.go#L52)?

Full disclosure: I'm not entirely sure that function is doing the same thing you are here, but I usually look for alternatives when I see something like `_, err = iso.ReadAt(volumeId, 32808)` :sweat_smile:

Reply:
I think I could add a more detailed comment about the bytes that we are skipping and why. But based on my reading of different implementations of iso readers, the location of the 32 bytes for the volume id is pretty fixed. I'm personally hesitant to bring in a whole iso package just for this.

WDYT about my explaining exactly what is happening in a comment.

For reference, this is where the volume id is being set when RHCOS images are built https://github.com/coreos/coreos-assembler/blob/master/src/cmd-buildextend-live#L103

Reply:
Don't worry about bringing that package in - I'm already using it in a PR that is based on this one. :)
That said, I'm not sure the library has a method for getting the volume ID.  From my experience with it it's more for reading files/directories inside the ISO, whereas the volume ID is probably in some metadata area.  Might be wrong though.

Reply:
Since I don't see anything in that package that, at least in a straightforward way, gives me the volume id I want, I will comment the `ReadAt(volumeId, 32808)` to communicate where `32808` comes from.

Reply:
Added a comment to explain this. 32768 bytes are unused and volume id is 40 bytes in. Let me know if the comment is sufficient.

https://wiki.osdev.org/ISO_9660
https://en.wikipedia.org/wiki/ISO_9660